tracing-core/src/field.rs

```diff
@@ -225,6 +225,17 @@ pub trait Visit {
 
     /// Visit a value implementing `fmt::Debug`.
     fn record_debug(&mut self, field: &Field, value: &dyn fmt::Debug);
+
+    /// Visit an Option
+    fn record_option(&mut self, field: &Field, value: Option<&(dyn Value + 'static)>)
+    where
+        Self: std::marker::Sized,
```

Comment:
You *don't* want to have a `Self: Sized` constraint here because this trait is meant to be used with dynamic dispatch.

If you added it because of the borrow checker try using a reborrow (`&mut *self`) to pass `self` to `inner_value.record()`.